File: src/HW1/Task7.java
//Декларативное программирование:
//Задача 7: Поиск подстроки.
// Напишите программу, которая находит все вхождения заданной подстроки в строке с использованием встроенных функций.

package HW1;

import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class Task7 {
    public static void main(String[] args) {
        String textString = "Hello world from Java";
        String substring = "a";

        Pattern p = Pattern.compile(substring);
        Matcher m = p.matcher(textString);

        while (m.find()) {
            System.out.println(m.group());
        }
    }
}

File: src/HW1/task_7.py
# Декларативное программирование:
# Задача 7: Поиск подстроки.
# Напишите программу, которая находит все вхождения заданной подстроки в строке с использованием встроенных функций.

text = 'Hello world from Java'
sub = 'a'

result = [text[i] for i in range(len(text)) if text.startswith(sub, i)]

print(result)
